kernel/ata.c

```diff
@@ -409,14 +409,16 @@ int ata_probe( int id, int *nblocks, int *blocksize, char *name )
 	cbuffer[256]=0;
 
 	/* Vendor supplied name is at byte 54 */
-
 	strcpy(name,&cbuffer[54]);
 	name[40] = 0;
 
-	console_printf("ata unit %d: %s %d MB %s\n",
+	/* Get disk size in megabytes*/
+	double mbytes = ((double) *nblocks) * (((double) *blocksize)/MEGA);
```

Comment:
Avoid the use of floating point in kernel code -- it can result in some strange issues related to if/when the FPU state is initialized and recovered.  Fix this by doing integer math in a different order.

Reply:
See 
https://github.com/dthain/basekernel/pull/164/commits/4011498d95044c1cb8d031ae9e04c0eacf75348f